packages/account/tests/account.e2e.spec.ts

```diff
@@ -456,4 +457,114 @@ describe("Account Tests", () => {
 
     expect(ecdsaOwnershipModule).toBe(smartAccount.activeValidationModule.getAddress());
   });
+
+  it("should deploy a smart account with native token balance", async () => {
+    const {
+      bundlerUrl,
+      biconomyPaymasterApiKey,
+      viemChain,
+      publicClient,
+      whale: { viemWallet: signer, account },
+      deploymentCost,
+    } = mumbai;
+
+    const newPrivateKey = generatePrivateKey();
+    const newAccount = privateKeyToAccount(newPrivateKey);
+
+    const newViemWallet = createWalletClient({
+      account: newAccount,
+      chain: viemChain,
+      transport: http(viemChain.rpcUrls.default.http[0]),
+    });
+
+    const smartAccount = await createSmartAccountClient({
+      signer: newViemWallet,
+      biconomyPaymasterApiKey,
+      bundlerUrl,
+    });
+
+    const smartAccountAddress = await smartAccount.getAccountAddress();
+
+    // Setup:
+    const hash = await signer.sendTransaction({ to: smartAccountAddress, value: BigInt(deploymentCost), account, chain: viemChain }); // Send enough native token to counterfactual address to deploy the smart account
+    const transaction = await publicClient.waitForTransactionReceipt({ hash });
+    expect(transaction).toBeTruthy();
+
+    // Test:
+    const { wait } = await smartAccount.deploy();
+    const { success } = await wait();
+
+    const byteCode = await publicClient.getBytecode({ address: smartAccountAddress });
+    expect(success).toBe("true");
+    expect(byteCode).toBeTruthy();
+  }, 60000);
+
+  it("should deploy a smart account with sponsorship", async () => {
+    const { bundlerUrl, biconomyPaymasterApiKey, viemChain, publicClient } = mumbai;
+
+    const newPrivateKey = generatePrivateKey();
+    const newAccount = privateKeyToAccount(newPrivateKey);
+
+    const newViemWallet = createWalletClient({
+      account: newAccount,
+      chain: viemChain,
+      transport: http(viemChain.rpcUrls.default.http[0]),
+    });
+
+    const smartAccount = await createSmartAccountClient({
+      signer: newViemWallet,
+      biconomyPaymasterApiKey,
+      bundlerUrl,
+    });
+
+    const smartAccountAddress = await smartAccount.getAccountAddress();
+    const balance = await publicClient.getBalance({ address: smartAccountAddress });
+    expect(balance).toBe(0n);
+
+    const { wait } = await smartAccount.deploy({
+      paymasterServiceData: { mode: PaymasterMode.SPONSORED },
+    });
+    const { success } = await wait();
+
+    const byteCode = await publicClient.getBytecode({ address: smartAccountAddress });
```

Comment:
in my knowledge it would never come undefined. either '0x' or some real bytecode (basically length greater than 2)

Reply:
I found it to be undefined. I think you're describing a difference between ethers and viem

Reply:
[This tests passes](Yes, [here](https://github.com/bcnmy/biconomy-client-sdk/pull/429/files/8c3ebd512ee491806d7e79ce0c61bae71c2df621#diff-f9b56b02ea191c9a2bf7380a0dd2702bd35827ffefc83014b18bf2d03a904f76R568))